docs/ref/contrib/admin/filters.txt

```diff
@@ -0,0 +1,188 @@
+            provided in the query string and retrievable via
+            `self.value()`.
+            """
+            # Compare the requested value (either '80s' or '90s')
+            # to decide how to filter the queryset.
+            if self.value() == '80s':
+                return queryset.filter(birthday__gte=date(1980, 1, 1),
+                                        birthday__lte=date(1989, 12, 31))
+            if self.value() == '90s':
+                return queryset.filter(birthday__gte=date(1990, 1, 1),
+                                        birthday__lte=date(1999, 12, 31))
+
+    class PersonAdmin(admin.ModelAdmin):
+        list_filter = (DecadeBornListFilter,)
+
+.. note::
+
+    As a convenience, the ``HttpRequest`` object is passed to the ``lookups``
+    and ``queryset`` methods, for example::
+
+        class AuthDecadeBornListFilter(DecadeBornListFilter):
+
+            def lookups(self, request, model_admin):
+                if request.user.is_superuser:
+                    return super().lookups(request, model_admin)
+
+            def queryset(self, request, queryset):
+                if request.user.is_superuser:
+                    return super().queryset(request, queryset)
+
+    Also as a convenience, the ``ModelAdmin`` object is passed to the
+    ``lookups`` method, for example if you want to base the lookups on the
+    available data::
+
+        class AdvancedDecadeBornListFilter(DecadeBornListFilter):
+
+            def lookups(self, request, model_admin):
+                """
+                Only show the lookups if there actually is
+                anyone born in the corresponding decades.
+                """
+                qs = model_admin.get_queryset(request)
+                if qs.filter(birthday__gte=date(1980, 1, 1),
+                            birthday__lte=date(1989, 12, 31)).exists():
+                    yield ('80s', _('in the eighties'))
+                if qs.filter(birthday__gte=date(1990, 1, 1),
+                            birthday__lte=date(1999, 12, 31)).exists():
+                    yield ('90s', _('in the nineties'))
+
+Using a field name and an explicit ``FieldListFilter``
+======================================================
+
+Finally, if you wish to specify an explicit filter type to use with a field you
+may provide a ``list_filter`` item as a tuple, where the first element is a
+field name and the second element is a class inheriting from
+``django.contrib.admin.FieldListFilter``, for example::
+
+    class PersonAdmin(admin.ModelAdmin):
+        list_filter = (
+            ('is_staff', admin.BooleanFieldListFilter),
+        )
+
+Here the ``is_staff`` field will use the ``BooleanFieldListFilter``. Specifying
+only the field name, fields will automatically use the appropriate filter for
+most cases, but this format allows you to control the filter used.
+
+There are some available classes that will not be used automatically.
```

Comment:
As a reader, my interest is piqued - why, and what are they? Are they secret/private, or just opt-in? Are they enumerated further down (It doesn't _read_ like that's the case, but may the subsequent ones are the very same)?

Reply:
They are enumerated below. I'll try to clarify. 🧐